python/dolfinx/fem/element.py

```diff
@@ -189,14 +189,14 @@ def dtype(self) -> np.dtype:
         """Geometry type of the Mesh that the FunctionSpace is defined on."""
         return self._cpp_object.dtype
 
-    @property
+    @cached_property
```

Comment:
Cached property should be signposted in docstring as it adds hidden state to the method.